Upgrade inquirer version to 11.0.2

Aad1tya27 opened the pull request:
**What kind of change does this PR introduce?**
The version of 'inquirer' package has been upgraded to 11.0.2 
Introduced Validation in ```askForCustomPort.ts``` 
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #2294  <!--Add related issue number here.-->

**Did you add tests for your changes?**
I edited ```src/setup/askForCustomPort/askForCustomPort.test.ts```  and ```src/setup/askForTalawaApiUrl/askForTalawaApiUrl.test.ts```
<!--Yes or No. Note: Add unit tests or automation tests for your code.-->

**Snapshots/Videos:**
![image](https://github.com/user-attachments/assets/97137e06-2192-4645-9f97-f274d4377a86)
![image](https://github.com/user-attachments/assets/59ba7dca-334a-4185-97e3-3f17e2a67d56)
![image](https://github.com/user-attachments/assets/85809139-1cda-4a05-9af8-14b0b45e80ba)

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**
No
<!--Add link to Talawa-Docs.-->

**Summary**
1. Updated the version of 'inquirer' package.
2. Faced bugs when returning a number value as default, so I set the default value as a string and converted to number before returning the custom port.
3. Fixed test where expected value was wrong.
4. Fixed test which was giving error due to types, by giving the mockValue the default endpoint ```http://localhost:4000/graphql```
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
Not sure
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->


**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes
<!--Yes or No-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Updated dependency on the `inquirer` package for potential improvements.
	- Enhanced input handling for custom port prompts, including validation and user confirmation for reserved ports.
	- Improved type safety for the Talawa API URL prompt.

- **Bug Fixes**
	- Adjusted expected output types in test cases to reflect changes in data handling for custom port input.

- **Tests**
	- Simplified mock behavior in tests for the Talawa API URL prompt.
	- Added new test cases for validating custom port inputs and handling reserved ports.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes commented:
1. Please make CodeRabbit.ai approve your PR
2. Make sure to cover your additions with valid tests
    ![image](https://github.com/user-attachments/assets/fcab8960-18c7-4856-8e74-ab985645f73b)

palisadoes reviewed (changes requested):
1. Please make CodeRabbit.ai approve your PR
2. Make sure to cover your additions with valid tests
    ![image](https://github.com/user-attachments/assets/fcab8960-18c7-4856-8e74-ab985645f73b)

Aad1tya27 commented on src/setup/askForCustomPort/askForCustomPort.test.ts:
```diff
@@ -19,6 +19,28 @@ describe('askForCustomPort', () => {
+    );
+  });
+
+  test('should handle invalid port input and prompt again', async () => {
+    jest
+      .spyOn(inquirer, 'prompt')
+      .mockResolvedValueOnce({ customPort: 'abcd' })
+      .mockResolvedValueOnce({ customPort: '8080' });
+
+    const result = await askForCustomPort();
+    expect(result).toBe(8080);
```
I have added test cases to test every function.

Aad1tya27 commented on src/setup/askForCustomPort/askForCustomPort.ts:
```diff
@@ -1,14 +1,33 @@
+    validPort = Number(customPort);
+    if (Number.isNaN(validPort) || validPort <= 0 || validPort > 65535) {
+      validPort = null;
+    }
```
I removed that part

Aad1tya27 commented on src/setup/askForCustomPort/askForCustomPort.ts:
```diff
@@ -1,14 +1,21 @@
+          return 'Please enter a valid port number between 1 and 65535.';
+        }
+        return true;
+      },
```
I have added that functionality

Aad1tya27 commented:
@palisadoes  I have added tests related to all functionalities of my code . The only job I'm failing is cause I changed ```package.lock.json``` to upgrade the inquirer package.
coderabbitai has approved my changes.
